tests/models/mistral/test_modeling_mistral.py

```diff
@@ -627,3 +627,127 @@ def test_speculative_generation(self):
         del model
         backend_empty_cache(torch_device)
         gc.collect()
+
+
+@slow
+@require_torch_gpu
+class Mask4DTestHard(unittest.TestCase):
```

Comment:
this test is the same as llama , let's use copied from

Reply:
comparing the test code for Llama and Mistral I see that:
- specific model used in setup()
- test examples are text-based (may not lead to same token count with different tokenizers)
- Llama version has extra methods to test StaticCache

Potential solutions:
0) do nothing
1) get away from text-based examples and use token_ids for transferability across models
  1a) copy tests from Llama to Mistral
  1b) move the Mask4DTestHard class methods to testing_common

Reply:
yes no worries